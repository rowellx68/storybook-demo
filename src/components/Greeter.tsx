/** @jsx jsx */
import React from 'react';
import { css, jsx } from '@emotion/core';

import { headers } from 'styles/text';

const greeterStyles = css({
  marginBottom: '1rem',
  ...headers.h1,
});

const timeOfDay = (hour: number): string => {
  if (hour <= 11) {
    return 'morning';
  } else if (hour <= 16) {
    return 'afternoon';
  }

  return 'evening';
};

interface GreeterProps {
  name: string;
  subtitle?: string;
}

const Greeter: React.FunctionComponent<GreeterProps> = ({ name, subtitle }) => {
  const date = new Date();

  return (
    <div css={greeterStyles}>
      <h1>
        Good {timeOfDay(date.getHours())}, {name}!
      </h1>
      {subtitle ? <h2>{subtitle}</h2> : null}
    </div>
  );
};

export default Greeter;
