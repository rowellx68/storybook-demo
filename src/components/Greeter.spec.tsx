import React from 'react';
import { mount } from 'enzyme';
import FakeTimers from '@sinonjs/fake-timers';

import Greeter from 'components/Greeter';

describe('Greeter', () => {
  it('renders the correct greeting - morning', () => {
    const clock = FakeTimers.install({ now: new Date(1577876400000) });

    const component = mount(<Greeter name='John' />);

    const h1 = component.find('h1');

    const expected = <h1>Good morning, John!</h1>;

    expect(h1).toContainReact(expected);
    expect(component).toMatchSnapshot();

    clock.uninstall();
  });

  it('renders the correct greeting - afternoon', () => {
    const clock = FakeTimers.install({ now: new Date(1577890800000) });

    const component = mount(<Greeter name='John' />);

    const h1 = component.find('h1');

    const expected = <h1>Good afternoon, John!</h1>;

    expect(h1).toContainReact(expected);
    expect(component).toMatchSnapshot();

    clock.uninstall();
  });

  it('renders the correct greeting - evening', () => {
    const clock = FakeTimers.install({ now: new Date(1577908800000) });

    const component = mount(<Greeter name='John' />);

    const h1 = component.find('h1');

    const expected = <h1>Good evening, John!</h1>;

    expect(h1).toContainReact(expected);
    expect(component).toMatchSnapshot();

    clock.uninstall();
  });
});
