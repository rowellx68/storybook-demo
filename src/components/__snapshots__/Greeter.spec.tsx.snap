// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeter renders the correct greeting - afternoon 1`] = `
.emotion-0 {
  margin-bottom: 1rem;
  font-size: 1.7rem;
}

@media screen and (min-width:640px) {
  .emotion-0 {
    font-size: 2rem;
  }
}

<Greeter
  name="John"
>
  <div
    className="emotion-0"
  >
    <h1>
      Good 
      afternoon
      , 
      John
      !
    </h1>
  </div>
</Greeter>
`;

exports[`Greeter renders the correct greeting - evening 1`] = `
.emotion-0 {
  margin-bottom: 1rem;
  font-size: 1.7rem;
}

@media screen and (min-width:640px) {
  .emotion-0 {
    font-size: 2rem;
  }
}

<Greeter
  name="John"
>
  <div
    className="emotion-0"
  >
    <h1>
      Good 
      evening
      , 
      John
      !
    </h1>
  </div>
</Greeter>
`;

exports[`Greeter renders the correct greeting - morning 1`] = `
.emotion-0 {
  margin-bottom: 1rem;
  font-size: 1.7rem;
}

@media screen and (min-width:640px) {
  .emotion-0 {
    font-size: 2rem;
  }
}

<Greeter
  name="John"
>
  <div
    className="emotion-0"
  >
    <h1>
      Good 
      morning
      , 
      John
      !
    </h1>
  </div>
</Greeter>
`;
